### PR TITLE
dkg: add frost command test

### PR DESCRIPTION
**Pull request by corverroos**
Adds frost algo to the dkg command test.
- Add public shares to cluster lock file.

category: test
ticket: #478 
feature_set: alpha

### CODE REVIEW

File: cluster/distvalidator.go [hunk 1]
```diff
@@ -22,8 +22,13 @@ type DistValidator struct {
 	// PubKey is the root distributed public key.
 	PubKey string `json:"distributed_public_key"`
 
-	// Verifiers are the public shares.
-	Verifiers [][]byte `json:"threshold_verifiers"`
+	// PublicShares are the public keys of each node's secret key share.
+	// It can be used to verify a partial signature created by any node in the cluster.
+	PublicShares [][]byte `json:"public_shares,omitempty"`
```
Comment (xenowits) [inline cluster/distvalidator.go:27]: nit: can name this `PubShares`

File: cluster/distvalidator.go [hunk 2]
```diff
@@ -22,8 +22,13 @@ type DistValidator struct {
 	// PubKey is the root distributed public key.
 	PubKey string `json:"distributed_public_key"`
 
-	// Verifiers are the public shares.
-	Verifiers [][]byte `json:"threshold_verifiers"`
+	// PublicShares are the public keys of each node's secret key share.
```
Comment (xenowits) [inline cluster/distvalidator.go:25]:
```suggestion
	// PubShares are the public keys corresponding to each node's secret key share.
```

File: cluster/distvalidator.go [hunk 3]
```diff
@@ -22,8 +22,13 @@ type DistValidator struct {
...
+	// It can be used to verify a partial signature created by any node in the cluster.
+	PublicShares [][]byte `json:"public_shares,omitempty"`
+
+	// Verifiers are the threshold verifier commitments.
+	// Deprecated: Use PublicShares.
+	Verifiers [][]byte `json:"threshold_verifiers,omitempty"`
```
Comment (xenowits) [inline cluster/distvalidator.go:31]: discussion: why are we keeping this around then?

File: dkg/dkg_test.go [hunk 4]
```diff
@@ -108,17 +125,17 @@ func TestDKG(t *testing.T) {
 
 	// Read generated lock and keystores from disk
 	var (
-		shares = make([][]*bls_sig.SecretKey, vals)
-		locks  []cluster.Lock
+		secretShares = make([][]*bls_sig.SecretKey, def.NumValidators)
+		locks        []cluster.Lock
```
Comment (xenowits) [inline dkg/dkg_test.go:129]:
```suggestion
		locks []cluster.Lock
```

File: tbls/tblsconv/tblsconv.go [hunk 5]
```diff
@@ -26,6 +26,16 @@ import (
 	"github.com/obolnetwork/charon/core"
 )
 
+// KeyFromBytes unmarshalls the bytes into a kryptology bls public key.
```
Comment (xenowits) [inline tbls/tblsconv/tblsconv.go:29]:
```suggestion
// KeyFromBytes unmarshals the bytes into a kryptology bls public key.
```

File: dkg/dkg_test.go [hunk 6]
```diff
@@ -36,25 +37,40 @@ import (
...
+	})
+
+	t.Run("frost", func(t *testing.T) {
+		lock, keys, _ := cluster.NewForT(t, vals, nodes, nodes, 0)
+		lock.Definition.DKGAlgorithm = "frost"
+		testDKG(t, lock.Definition, keys)
+	})
+}
+
+func testDKG(t *testing.T, def cluster.Definition, p2pKeys []*ecdsa.PrivateKey) {
```
Comment (xenowits) [inline dkg/dkg_test.go:64]: nit: we can also have another test which checks the negative cases, like what happens when we have a malformed or incomplete definition/lock

**Review by xenowits — APPROVED** [review]

File: cluster/distvalidator.go (hunk 3 above)
Comment (corverroos) [reply cluster/distvalidator.go:31]: still used, will refactor in another PR

File: dkg/dkg_test.go (hunk 4 above)
Comment (corverroos) [reply dkg/dkg_test.go:129]: this was `gofmt`

File: dkg/dkg_test.go (hunk 6 above)
Comment (corverroos) [reply dkg/dkg_test.go:64]: you are welcome to add these in another PR

File: cluster/distvalidator.go (hunk 1 above)
Comment (corverroos) [reply cluster/distvalidator.go:27]: good idea, will rename and and fix types in another PR

Comment (corverroos) [reply cluster/distvalidator.go:27]: ok cool

File: cluster/distvalidator.go (hunk 2 above)
Comment (corverroos) [reply cluster/distvalidator.go:25]: ok cool

File: tbls/tblsconv/tblsconv.go (hunk 5 above)
Comment (corverroos) [reply tbls/tblsconv/tblsconv.go:29]: cool
